### PR TITLE
feat(datafusion): Add LIKE predicate pushdown for StartsWith patterns

### PR DESCRIPTION
**Pull request by viirya**
Implement pushdown for LIKE predicates with simple prefix patterns (e.g., 'prefix%') to Iceberg's StartsWith operator. This optimization allows filtering to be performed at the storage layer, significantly improving query performance for prefix searches.

Changes:
- Add support for Expr::Like in predicate conversion
- Convert LIKE 'prefix%' patterns to StartsWith operator
- Convert NOT LIKE 'prefix%' patterns to NotStartsWith operator
- Handle edge cases: empty prefix, unicode, special characters
- Reject complex patterns that cannot be pushed down (wildcards in middle, underscore, ILIKE)
- Add 8 comprehensive unit tests covering various scenarios

Implementation details:
- Only simple prefix patterns ending with % are converted
- Patterns with % or _ wildcards in the prefix are not pushed down
- Case-insensitive LIKE (ILIKE) is not supported for pushdown
- Escape characters are not supported for pushdown
- Works seamlessly with other predicates in AND/OR expressions

## Which issue does this PR close?

- Closes #.

### CODE REVIEW

File: crates/integrations/datafusion/src/physical_plan/expr_to_predicate.rs [hunk 1]
```diff
@@ -429,4 +477,97 @@ mod tests {
         let predicate = convert_to_iceberg_predicate(sql);
         assert_eq!(predicate, None);
     }
+
+    #[test]
```
Comment (liurenjie1024) [inline crates/integrations/datafusion/src/physical_plan/expr_to_predicate.rs:512]: We recently added support for sqllogictests, see https://github.com/liurenjie1024/iceberg-rust/blob/666a9fe1aaf1692583d6f44e4f7a1d52a688b217/crates/sqllogictest/testdata/schedules/df_test.toml#L19

It would be better if we also include such sql logic tests in spite of ut.

File: crates/integrations/datafusion/src/physical_plan/expr_to_predicate.rs [hunk 2]
```diff
@@ -128,6 +128,54 @@ fn to_iceberg_predicate(expr: &Expr) -> TransformedResult {
...
+        Expr::Like(Like {
+            negated,
+            expr,
+            pattern,
+            escape_char,
+            case_insensitive,
+        }) => {
+            // Only support simple prefix patterns (e.g., 'prefix%')
+            // Escape characters and case-insensitive matching are not supported yet
+            if escape_char.is_some() || *case_insensitive {
```
Comment (liurenjie1024) [inline crates/integrations/datafusion/src/physical_plan/expr_to_predicate.rs:142]: IIRC, iceberg's starts with is case sensitive.

File: crates/integrations/datafusion/src/physical_plan/expr_to_predicate.rs (hunk 1 above)
Comment (viirya) [reply crates/integrations/datafusion/src/physical_plan/expr_to_predicate.rs:512]: Okay. Added test in sqllogictests.

File: crates/integrations/datafusion/src/physical_plan/expr_to_predicate.rs (hunk 2 above)
Comment (viirya) [reply crates/integrations/datafusion/src/physical_plan/expr_to_predicate.rs:142]: I updated the document here.

**Comment by viirya** [comment]
Thanks @liurenjie1024
